pmd-core/src/main/java/net/sourceforge/pmd/RuleSetWriter.java

```diff
@@ -266,21 +265,11 @@ private Element createPropertiesElement(List<PropertyDescriptor<?>> propertyDesc
         Element propertiesElement = null;
         if (propertyDescriptors != null) {
 
-            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) { // For
-                // each
-                // provided
-                // PropertyDescriptor
-
-                if (propertyDescriptor instanceof PropertyDescriptorWrapper) { // Any
-                    // wrapper
-                    // property
-                    // needs
-                    // to
-                    // go
-                    // out
-                    // as
-                    // a
-                    // definition.
+            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) {
+                // For each provided PropertyDescriptor
+
+                if (propertyDescriptor instanceof PropertyDescriptorWrapper) {
```

Comment:
Maybe we could make the factories package private? Wouldn't that release the need to consider that factories can be used by rules/ downstream?